### PR TITLE
Delete tests.helpers.TrialMNISTDataModule

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Deletes the almost unused `TrialMNISTDataModule` in favor of `MNISTDataModule`
Checks if `sklearn` is available.

## Before submitting
- [n/a] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Comment by Borda** [comment]
I would personally keep it in codebase

**Comment by carmocca** [comment]
> I would personally keep it in codebase

I can add an option for the datamodule to use the trial-mnist dataset. But having both datamodules seems pointless, proved by the fact that it's almost unused.

**Comment by Borda** [comment]
> > I would personally keep it in codebase
> 
> I can add an option for the datamodule to use the trial-mnist dataset. But having both datamodules seems pointless, proved by the fact that it's almost unused.

ok, in such case, let's fuse both datasets into one datamodule

**Review by awaelchli — APPROVED** [review]

**Comment by carmocca** [comment]
> ok, in such case, let's fuse both datasets into one datamodule

@Borda done

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/helpers/datasets.py [hunk 1]
```diff
@@ -77,18 +77,15 @@ def __init__(
...
+        self.data, self.targets = self._try_load(os.path.join(self.cached_folder_path, data_file))
 
     def __getitem__(self, idx: int) -> Tuple[Tensor, int]:
         img = self.data[idx].float().unsqueeze(0)
         target = int(self.targets[idx])
 
-        if self.normalize is not None:
-            img = normalize_tensor(img, mean=self.normalize[0], std=self.normalize[1])
+        if self.normalize is not None and len(self.normalize) == 2:
+            img = self.normalize_tensor(img, **self.normalize)
```
Comment (carmocca) [inline tests/helpers/datasets.py:88]:
```suggestion
            img = self.normalize_tensor(img, *self.normalize)
```

File: tests/helpers/datamodules.py [hunk 2]
```diff
@@ -11,100 +11,51 @@
...
-
-    def test_dataloader(self):
-        return DataLoader(self.mnist_test, batch_size=32)
-
-    def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
-        checkpoint[self.__class__.__name__] = self.__class__.__name__
-
-    def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
-        self.checkpoint_state = checkpoint.get(self.__class__.__name__)
+_SKLEARN_AVAILABLE = _module_available("sklearn")
```
Comment (Borda) [inline tests/helpers/datamodules.py:23]: move this to `tests/__init__` as it could be useful elsewhere?

**Review by Borda — APPROVED** [review]

File: tests/helpers/datamodules.py (hunk 2 above)
Comment (carmocca) [reply tests/helpers/datamodules.py:23]: Not a bad idea, but would rather do it in a separate PR
